routes/car.js

```diff
@@ -1,8 +1,90 @@
 const express = require('express');
 const router = express.Router();
+const Car = require('../models/car');
+const User = require('../models/user');
+const jwtDecode = require('jwt-decode');
 
 router.get('/', (req, res) => {
-  res.send('car Route is working')
+    res.send('car Route is working')
 });
 
+
+router.post('/add', async (req, res) => {
+    const {_id} = jwtDecode(req.header('authorization'))['user'];
+
+    await User.findOne({_id: _id}).then(async (user) => {
+        if(user !== null || user !== 'undefined'){
+            await Car.findOne({createdBy: _id, license: req.body.license}).then(async (car) => {
+                if (car === null){
+                    const newCar = new Car(Object.assign({createdBy: _id}, req.body))
+                    await newCar.save().then((car) => {
+                        console.log('New car saved')
+                        user.cars.push(car._id)
+                        user.save()
+                        res.json(req.body)
+                    })
+                } else {
+                    console.log("Car already exists")
+                    res.status(409).json(req.body)
+                }
+            })
+        }
+    })
+})
+
+
```

Comment:
Same for other endpoints, try to avoid nested call backs
Instead use async await in a try/catch block
or using chaining with .then